src/main/java/com/example/feature/level1/controller/Level1Controller.java

```diff
@@ -0,0 +1,77 @@
+package com.example.feature.level1.controller;
+
+import com.example.feature.level1.domain.Posts;
+import com.example.feature.level1.domain.Users;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.Objects;
+import java.util.Random;
+
+/**
+ * Level1Applicationのコントローラクラス
+ */
+
+@Controller
+public class Level1Controller {
+    /**
+     * RestTemplateオブジェクト
+     */
+    private static final RestTemplate restTemplate = new RestTemplate();
+    /**
+     * JSONPlaceholder postsのベースURL、IDなし
+     */
+    private static final String baseUrl = "http://jsonplaceholder.typicode.com/posts/";
+    /**
+     * JSONPlaceholder usersのベースURL、IDなし
+     */
+    private static final String baseUrl2 = "http://jsonplaceholder.typicode.com/users/";
+    /**
+     * URLにランダムでIDを付加する用のRandomオブジェクト
+     */
+    Random random = new Random();
+
+    /**
+     * /indexにGET要求があったときの動作
+     * <a href="https://jsonplaceholder.typicode.com/posts">JSONPlaceholder posts</a>のURLに1~100までのランダムなIDを追加して1回GET、
+     * 取得したJSONのデータをビューに渡す
+     * @return "index"
+     * @see <a href="https://jsonplaceholder.typicode.com/posts">JSONPlaceholder posts</a>
+     */
+    @GetMapping("/index")
+    public String index(Model model) {
+        int randomId = 1 + random.nextInt(100); // 1~100
+        String randomUrl = baseUrl + randomId;
+
+        Posts p1 = restTemplate.getForObject(randomUrl, Posts.class);
```

Comment:
`randomId`,`randomUrl`,`p1`のような、**一回値を代入したら、その後書き換える予定がない変数**
に関しては都度`final`をつけておくと良いです。

`final`をつけて変数を宣言すると、その変数に対して再代入しようとするとエラーになるので
想定していない値の書き換えを防ぐことができます。
```suggestion
        final Posts p1 = restTemplate.getForObject(randomUrl, Posts.class);
```

Reply:
ありがとうございます！再代入をするかどうかの考えが抜けていました！

Reply:
小規模なWebアプリケーションや、一人での開発ではあまり効果を実感できないかもですが、
大規模なWebアプリケーションやチーム開発ではかなり効果のある方法なので覚えておいて損はないかと思います 👍